docs/proposals/rollout/rollout.md

```diff
@@ -0,0 +1,531 @@
+    // Workload of type deployment or daemonSet.
+    Workload *WorkloadReference `json:"workload"`
+
+    // ServiceName holds the name of a service which matches the `Workload`.
+    ServiceName string `json:"serviceName"`
+
+    // Port of the workload's Service which traffic access.
+    Port int32 `json:"port"`
+
+    // Primary is the labels and annotations to add to the primary service.
+    // Primary service is stable service.
+    // +optional
+    Primary *CustomMetadata `json:"primary,omitempty"`
+
+    // Canary is the labels and annotations to add to the canary service.
+    // Canary service is preview service.
+    // +optional
+    Preview *CustomMetadata `json:"preview,omitempty"`
+
+    // RolloutPolicy defines the Release Strategy of workload.
+    RolloutPolicy *RolloutPolicy `json:"rolloutPolicy"`
+}
+```
+
+`Testloader` indicates whether the user wants to install the test traffic load themselves. If you don't want to install the Testloader yourself, Kurator will install flagger's Testloader by default.
+
+`RolloutPolicy` defines the Rollout configuration for this installation workload. Although there is no detailed distinction in Kurator between canary, A/B testing and blue-green, giving users the freedom to configure traffic rules. Complete the release test. However, it is not allowed to configure canary and A/B or blue-green for the same workload.
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type RolloutPolicy struct {
+    // Checknum defines the number of checks to run for A/B Testing and Blue/Green
+    // Note: Kurator determines whether blue-green or A/B related processing is required based on 
+    // the presence or absence of content in the Checknum field. 
+    // So can't configure Iterations and CanaryStrategy at the same time.
+    // +optional
+    CheckNum int `json:"checknum,omitempty"`
+
+    // The TrafficRouting defines the configuration of the gateway, traffic distribution rules, and so on.
+    TrafficRouting *TrafficRoutingConfig `json:"trafficRouting"`
+
+    // TrafficAnalysis defines the validation process of a release
+    TrafficAnalysis *TrafficAnalysis `json:"trafficAnalysis,omitempty"`
+
+    // ProgressDeadlineSeconds represents the maximum time in seconds for a
+    // canary deployment to make progress before it is considered to be failed.
+    // Defaults to 600s.
+    // +optional
+    ProgressDeadlineSeconds *int32 `json:"progressDeadlineSeconds,omitempty"`
+
+    // SkipAnalysis promotes the canary without analyzing it
+    // +optional
+    SkipTrafficAnalysis bool `json:"skipTrafficAnalysis,omitempty"`
+
+    // Restore resources to initial state when deleting canary resources.
+    // Use of the revertOnDeletion property should be enabled
+    // when you no longer plan to rely on Kurator for deployment management.
+    // +optional
+    RevertOnDeletion bool `json:"revertOnDeletion,omitempty"`
```

Comment:
Can you explain how can we delete a canary resource within our fleet

Reply:
Kurator will install the Flagger to the specified cluster via a fleet plugin.  If RevertOnDeletion is set to true, the Flagger will revert a resource to its initial state  when the deleting rollout resource.

Reply:
Application.Spec.ApplicationSyncPolicy.Rollout field is cleared, or the application is deleted.